src/plugins/spectrogram.ts

```diff
@@ -728,12 +728,8 @@ class SpectrogramPlugin extends BasePlugin<SpectrogramPluginEvents, SpectrogramP
         const oldEnd = oldStart + oldPiece
         const newStart = i * newPiece
         const newEnd = newStart + newPiece
+        const overlap = Math.max(0, Math.min(oldEnd, newEnd) - Math.max(oldStart, newStart));
```

Comment:
```suggestion
        const overlap = Math.max(0, Math.min(oldEnd, newEnd) - Math.max(oldStart, newStart))
```

Lint will complain otherwise (`yarn prettier`).